### ISSUE TITLE
Google style: Disallow comments enclosed in boxes

### ISSUE THREAD

**Issue by VeniceWish** [checkstyle/checkstyle#17697]
I have read check documentation: https://checkstyle.sourceforge.io/checks/misc/todocomment.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

From: https://google.github.io/styleguide/javaguide.html#s4.8.6-comments
> Comments are not enclosed in boxes drawn with asterisks or other characters.

Code: 
```java
/*
  #################################
  #      Illegal Comment1         #
  #################################
*/
class Test {
  /**********************************
   *      Illegal Comment2          *
   **********************************/
  void doSomething(int x) {
    int y = x + 2;
    // -------------------
    //      comment
    // -------------------
  }

  // ======== separator ========
  void func() {}

  // ######
}

```
Cli: 
```cmd
$ java -jar checkstyle-10.26.1-all.jar -c google_checks.xml Test.java 
Starting audit...
[WARN] Test.java:7: First sentence of Javadoc is missing an ending period. [SummaryJavadoc]
Audit done.
```

Formatter:
```cmd
$ java -jar google-java-format-1.28.0-all-deps.jar Test.java > FormattedCode.java
$ diff -Naru Test.java FormattedCode.java
None
```

---

We need a new check to enforce that comments are not enclosed in boxes drawn with asterisks or other characters, as required by the [Google Java Style Guide §4.8.6](https://google.github.io/styleguide/javaguide.html#s4.8.6-comments).
Currently, Checkstyle provides [TodoComment](https://checkstyle.sourceforge.io/checks/misc/todocomment.html#TodoComment), which can be configured with a regex like:
```xml
<module name="Checker">
  <module name="TreeWalker">
    <module name="TodoComment">
      <property name="format" value="(?m)^\s*([*=_#-])\1{5,}\s*$"/>
    </module>
  </module>
</module>
```
Output:
```
$ java -jar checkstyle-10.26.1-all.jar -c config.xml Test.java
Starting audit...
[ERROR] Test.java:1:3: Comment matches to-do format '(?m)^\s*([*=_#-])\1{5,}\s*$'. [TodoComment]
[ERROR] Test.java:7:5: Comment matches to-do format '(?m)^\s*([*=_#-])\1{5,}\s*$'. [TodoComment]
[ERROR] Test.java:12:7: Comment matches to-do format '(?m)^\s*([*=_#-])\1{5,}\s*$'. [TodoComment]
[ERROR] Test.java:14:7: Comment matches to-do format '(?m)^\s*([*=_#-])\1{5,}\s*$'. [TodoComment]
[ERROR] Test.java:20:5: Comment matches to-do format '(?m)^\s*([*=_#-])\1{5,}\s*$'. [TodoComment]
Audit done.
Checkstyle ends with 5 errors.
```
This produces a similar effect, but `TodoComment` can only detect lines that match a regex.
It cannot determine whether those lines actually enclose a comment block.
As a result, even a single separator line like `// ######` is flagged as a violation.
**Note: Whether a single-line separator such as `// ######` should be considered a violation is debatable — this ultimately depends on the design decision of the new check.**

---

**Comment by romani** [comment]
@Tieshiyang , if you update your config to override message https://checkstyle.sourceforge.io/config.html#Custom_messages , Check violations can not reference "to-do" so message can be more clear.

I think we can put your Check instance to config and mention in coverage table that we do partial coverage for this requirement. 
In my mind detection of exactly a box is too much, detection of header of box should be enough. 
If users start complaining, we can remove it and as user for more details in what cases they see boxes and not boxes

**Comment by romani** [comment]
@mohitsatr , @Zopsss  are you ok with proposed solution?

**Comment by Zopsss** [comment]
I am fine with it. Detecting if the comment is a whole box or not can become too complex, not sure if it is possible for us or not. But for now, we can use this work around. If users complains about it then we will remove it.

**Comment by romani** [comment]
@Tieshiyang , issue is approved , you can send PR, thanks a lot for your help

**Comment by smita1078** [comment]
I am on it.
Changes I will make to original regex:
Focused character set `[*=#-]` : Targets the most common box-drawing characters (*, =, #, -)
Exclude underscore `_`: Less commonly used for boxes, removing it reduces false positives
Higher threshold `{8,}`: Requires 9+ character repetitions (typical boxes use 10+, while short separators use fewer)

**Comment by romani** [comment]
Fix is merged
